src/run/handlers/tags-handler.cts

```diff
@@ -0,0 +1,146 @@
+import { purgeCache } from '@netlify/functions'
+
+import { name as nextRuntimePkgName, version as nextRuntimePkgVersion } from '../../../package.json'
+import { TagManifest } from '../../shared/blob-types.cjs'
+import {
+  getMemoizedKeyValueStoreBackedByRegionalBlobStore,
+  MemoizedKeyValueStoreBackedByRegionalBlobStore,
+} from '../storage/storage.cjs'
+
+import { getLogger, getRequestContext } from './request-context.cjs'
+
+const purgeCacheUserAgent = `${nextRuntimePkgName}@${nextRuntimePkgVersion}`
+
+/**
+ * Get timestamp of the last revalidation for a tag
+ */
+async function lastTagRevalidationTimestamp(
+  tag: string,
+  cacheStore: MemoizedKeyValueStoreBackedByRegionalBlobStore,
+): Promise<number | null> {
+  const tagManifest = await cacheStore.get<TagManifest>(tag, 'tagManifest.get')
+  if (!tagManifest) {
+    return null
+  }
+  return tagManifest.revalidatedAt
```

Comment:
just dreaming - If we used `undefined` instead of `null` we could do `tagManifest?.revalidatedAt` and write less code

Reply:
I don't think producing less code lines is good goal - I think making the code the most readable is